crates/worker/src/partition_processor_manager.rs

```diff
@@ -155,6 +156,16 @@ pub enum Error {
     InvokerBuild(#[from] BuildError),
 }
 
+enum RestartDelay {
```

Comment:
A reasonable critique would be that these add a lot of verbosity when we could just be passing delays around. I personally find the code _much_ easier to read with explicit names.

Since we don't have a single retry policy that can be used, and we need to track some per-partition state, it's not possible to reuse the existing `RetryPolicy` implementation here. This enables us to shift the delay calculation out of the already large async event handler.